auth/service.go

```diff
@@ -95,11 +94,23 @@ type Service interface {
 	PATS
 }
 
+//go:generate mockery --name Cache --output=./mocks --filename cache.go --quiet --note "Copyright (c) Abstract Machines"
+type Cache interface {
+	Save(ctx context.Context, scopes []Scope) error
+
+	CheckScope(patID, optionalDomainID string, entityType EntityType, operation Operation, entityID string) bool
+
```

Comment:
Add `context.Context` to `CheckScope`

```suggestion
	CheckScope(ctx context.Context, patID, optionalDomainID string, entityType EntityType, operation Operation, entityID string) bool
```